src/containers/dashboard/CrisisReportDashboardContainer.js

```diff
@@ -0,0 +1,204 @@
+/*
+ * @flow
+ */
+
+import React from 'react';
+
+import styled from 'styled-components';
+import { Map } from 'immutable';
+import { Spinner } from 'lattice-ui-kit';
+import { connect } from 'react-redux';
+import { bindActionCreators } from 'redux';
+import type { RequestSequence } from 'redux-reqseq';
+
+import { loadDashboardData } from './DashboardActionFactory';
+import { SUBMISSION_STATES } from './DashboardReducer';
+
+import ButtonToolbar from '../../components/buttons/ButtonToolbar';
+import DropdownButton from '../../components/buttons/DropdownButton';
+import IncidentCharts from '../../components/dashboard/IncidentCharts';
+import OutcomeCharts from '../../components/dashboard/OutcomeCharts';
+import OverviewCharts from '../../components/dashboard/OverviewCharts';
+import SummaryStats from '../../components/dashboard/SummaryStats';
+import { SUMMARY_STATS } from '../../shared/Consts';
+
+const OptionRow = styled.div`
+  display: flex;
+  flex-direction: row;
+  justify-content: space-between;
+  align-items: center;
+`;
+
+const Flex = styled.div`
+  display: flex;
+  flex: 1;
+  justify-content: center;
+  align-items: center;
+  width: 100%;
+`;
+
+const DashboardWrapper = styled.div`
+  width: 100%;
+`;
+
+const NoReports = styled.div`
+  margin-top: 50px;
+  width: 100%;
+  text-align: center;
+  font-size: 16px;
+  color: #2e2e34;
+`;
+
+const LAYOUTS = {
+  OVERVIEW: 'Overview',
+  INCIDENT: 'Incident',
+  OUTCOMES: 'Outcomes'
+};
+
+const TIME_LENGTHS = {
+  MONTHS_1: {
+    label: '1 Month',
+    value: 1
+  },
+  MONTHS_3: {
+    label: '3 Months',
+    value: 3
+  },
+  MONTHS_6: {
+    label: '6 Months',
+    value: 6
+  },
+  YEAR: {
+    label: '1 Year',
+    value: 12
+  }
+};
+
+type Props = {
+  app :Map,
+  dashboardCounts :Map,
+  isLoading :boolean,
+  summaryStats :Map,
+  actions :{
+    loadDashboardData :RequestSequence;
+  }
+};
+
+type State = {
+  layout :string;
+  timeRange :any;
+};
+
+class DashboardContainer extends React.Component<Props, State> {
```

Comment:
Please switch to a function components + hooks.

Reply:
this is actually legacy CARE code that i just copied to a new file—but i can do that to keep the project updated